indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -583,7 +584,9 @@ public static Path makeSegmentOutputPath(
       DataSegment segment
   )
   {
-    String segmentDir = "hdfs".equals(fileSystem.getScheme()) || "viewfs".equals(fileSystem.getScheme())
+    String segmentDir = "hdfs".equals(fileSystem.getScheme())
+                        || "viewfs".equals(fileSystem.getScheme())
+                        || "s3a".equals(fileSystem.getScheme())
```

Comment:
s3a shouldn't really be here… probably just like determining the loadSpec, the directory should be based on the kind of deep storage configured and not on the scheme. I think adding a getStorageDir to DataSegmentPusher, and getting rid of DataSegmentPusherUtil, would solve that. It could use java 8 default methods to prevent any pusher other than HDFS from having to override it.